tests/templates/kuttl/resources/10-assert.yaml

```diff
@@ -3,10 +3,82 @@ apiVersion: kuttl.dev/v1beta1
 kind: TestAssert
 timeout: 900
 ---
-# The Job starting the whole process
 apiVersion: spark.stackable.tech/v1alpha1
 kind: SparkApplication
 metadata:
   name: spark-resources
 status:
   phase: Succeeded
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    spark-role: driver
+    spark-app-name: spark-resources
+    app.kubernetes.io/managed-by: spark-k8s-operator
+spec:
+  containers:
+    - name: spark-driver
+      resources:
+        limits:
+          cpu: 1500m
+          # N.B. spark adds memoryOverhead of 10% by default
+          memory: 1408Mi
+        requests:
+          cpu: "1"
+          memory: 1408Mi
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    job-name: spark-resources
+    app.kubernetes.io/managed-by: spark-k8s-operator
+spec:
+  containers:
+    - name: spark-submit
+      resources:
+        limits:
+          cpu: 200m
+          memory: 1Gi
+        requests:
+          cpu: 100m
+          memory: 1Gi
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    spark-app-name: sparkals
```

Comment:
Is this correct ? Is spark overriding this label ?

Reply:
I don't think we set that (at least I couldn't find any reference to that in the code) - that seems to come from the main application class.

Reply:
I think you might be right: I had some issues with slightly different labels being assigned by KinD / managed k8s. I'll look into separately.